src/group_theory/subgroup.lean

```diff
@@ -589,6 +597,70 @@ as additive groups"]
 def prod_equiv (H : subgroup G) (K : subgroup N) : H.prod K ≃* H × K :=
 { map_mul' := λ x y, rfl, .. equiv.set.prod ↑H ↑K }
 
+/-- A subgroup is normal if whenever `n ∈ H`, then `g * n * g⁻¹ ∈ H` for every `g : G` -/
+@[class, to_additive "An add_subgroup is normal if whenever `n ∈ H`, then
+  `g + n - g ∈ H` for every `g : G` "] def normal : Prop :=
+∀ n, n ∈ H → ∀ g : G, g * n * g⁻¹ ∈ H
+
+variable {H}
+@[instance, to_additive] lemma normal_of_comm {G : Type*} [comm_group G] (H : subgroup G) : H.normal :=
+by simp [normal, mul_comm, mul_left_comm]
+
+namespace normal
+
+variable nH : H.normal
+
+@[to_additive] lemma conj_mem : ∀ n, n ∈ H → ∀ g : G, g * n * g⁻¹ ∈ H := nH
+
+@[to_additive] lemma mem_comm {a b : G} (h : a * b ∈ H) : b * a ∈ H :=
+have a⁻¹ * (a * b) * a⁻¹⁻¹ ∈ H, from nH (a * b) h a⁻¹, by simpa
+
+@[to_additive] lemma mem_comm_iff {a b : G} : a * b ∈ H ↔ b * a ∈ H :=
+⟨nH.mem_comm, nH.mem_comm⟩
+
+end normal
+
+@[instance, to_additive] lemma bot_normal : normal (⊥ : subgroup G) := by simp [normal]
+
+/-- The center of a group `G` is the set of elements that commute with everything in `G` -/
+variable (G)
+@[to_additive "The center of a group `G` is the set of elements that commute with everything in `G`"]
+def center : subgroup G :=
+{ carrier := {z | ∀ g, g * z = z * g},
+  one_mem' := by simp,
+  mul_mem' := λ a b (ha : ∀ g, g * a = a * g) (hb : ∀ g, g * b = b * g) g,
+    by assoc_rw [ha, hb g],
+  inv_mem' := λ a (ha : ∀ g, g * a = a * g) g,
+    by rw [← inv_inj', mul_inv_rev, inv_inv, ← ha, mul_inv_rev, inv_inv] }
+
+variable {G}
+
+@[to_additive] lemma mem_center_iff {z : G} : z ∈ center G ↔ ∀ g, g * z = z * g := iff.rfl
+
+@[instance, to_additive] lemma center_normal : (center G).normal :=
+begin
+  assume n hn g h,
+  assoc_rw [hn (h * g), hn g],
+  simp
+end
+
+variables {G} (H)
+/-- The `normalizer` of `H` is the smallest subgroup of `G` inside which `H` is normal. -/
+@[to_additive "The `normalizer` of `H` is the smallest subgroup of `G` inside which `H` is normal."]
+def normalizer : subgroup G :=
```

Comment:
Shouldn't there be an instance saying that this is normal?

Reply:
You mean an instance for `normal (H.comap (normalizer H).subtype)`?

Reply:
Euuh, I guess so. I was writing too fast, and confusing this with the normal closure.

Reply:
Your expression looks so complicated that I wonder if we should have a characteristic predicate for the normalizer.